whylog/config/parsers.py

```diff
@@ -57,3 +67,136 @@ def create_from_intent(cls, parser_intent):
+        )
+        return forward_regex, backward_regex
+
+    def _get_indexes_of_groups_for_parsers(self, parser_list):
+        indexes_dict = {}
+        free_index = 0
+        for parser in parser_list:
+            amount_of_group = parser.regex_str.count('(') - parser.regex_str.count('\(')
+            indexes_dict[parser.name] = (free_index, amount_of_group)
+            free_index += amount_of_group + 1
+        return indexes_dict
+
+    def _number_in_list(self):
+        regex_numbers = {}
+        number = 0
+        for parser in self._parsers:
+            regex_numbers[parser.name] = number
+            number += 1
+        return regex_numbers
+
+    def _create_group_index_to_regex_name(self, parsers_indexes):
+        index_to_regex = {}
+        for name, indexes in parsers_indexes.items():
+            index_to_regex[indexes[0]] = name
+        return index_to_regex
+
+    def get_extracted_regex_params(self, line):
+        if IMPORTED_RE:
+            clues = {}
+            self._brute_subregexes_matching(clues, 0, len(self._parsers) - 1, line)
+            return clues
+        forward_matched = self._forward_regex.match(line)
+        if forward_matched is None:
+            return ConcatedRegexParser.NO_MATCH
+        forward_groups = forward_matched.groups()
+        if self._is_matched_last_group(forward_groups):
+            return self._extract_params_from_last_regex(forward_groups)
+        backward_groups = self._backward_regex.match(line).groups()
+        forward_matched_regex_name, only_one = self._check_that_only_one_regex_matched(
+            forward_groups, backward_groups
+        )
+        regex_params = self._extract_regex_params_by_regex_name(
+            forward_groups, forward_matched_regex_name, self._forward_parsers_indexes
+        )
+        clues = {forward_matched_regex_name: regex_params}
+        if only_one:
+            return clues
+        backward_matched_regex_name = self._get_first_matches_regex_name(
+            backward_groups, self._backward_group_index_to_regex
+        )
+        regex_params = self._extract_regex_params_by_regex_name(
+            backward_groups, backward_matched_regex_name, self._backward_parsers_indexes
+        )
+        clues[backward_matched_regex_name] = regex_params
+        left = self._numbers_in_list[forward_matched_regex_name] + 1
+        right = self._numbers_in_list[backward_matched_regex_name] - 1
+        self._brute_subregexes_matching(clues, left, right, line)
+        return clues
+
+    def _extract_regex_params_by_regex_name(self, groups, matched_regex_name, parsers_indexes):
+        start_index = parsers_indexes[matched_regex_name][0]
+        number_of_groups = parsers_indexes[matched_regex_name][1]
+        regex_params = self._extract_regex_params(groups, number_of_groups, start_index)
+        return regex_params
+
+    def _get_first_matches_regex_name(self, groups, group_index_to_regex):
```

Comment:
s/_get_first_matches_regex_name/_get_first_matched_parser_name/ ?
s/group_index_to_regex/group_index_to_parser_name/ ?

this class keeps referring to regex params and regex inappropriately, please correct it so that it matches what is actually being passed between functions